cdap-test/src/main/java/co/cask/cdap/test/ApplicationManager.java

```diff
@@ -98,6 +98,12 @@
   void stopProgram(ProgramId programId);
 
   /**
+   * Wait for a given programId to have no running run record
+   * @param programId the program id to wait on
+   */
+  void waitForStopped(final ProgramId programId) throws Exception;
```

Comment:
The method `waitForStatus` already exists in `ProgramManager`. But no, it was added because `ApplicationManager#stopAll` is called, and that stops all program runs in the `stopAll`. It's best to have the method `waitForStopped` invoked only in `ApplicationManager` because that's called in the `afterTest`, and not involved in `ProgramManager`.